routes/characters.js

```diff
@@ -0,0 +1,134 @@
+var express = require('express');
+var router = express.Router();
+var mongoose = require('mongoose');
+
+var Character = require('../models/Character');
+
+router.get('/', (req, res, next) => {
+  Character.find()
+    .select('_id name level')
+    .exec()
+    .then(docs => {
+      var response = {
+        count: docs.length,
+        characters: docs.map(doc => {
+          return {
+            _id: doc._id,
+            name: doc.name,
+            level: doc.level,
+            request: {
+              type: 'GET',
+              url: 'http://localhost:3000/characters/' + doc._id
+            }
+          }
+        })
+      };
+      res.status(200).json(response);
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({ error: err });
+    });;
+});
+
+router.post('/', (req, res, next) => {
+  const character = new Character({
+
+    _id: new mongoose.Types.ObjectId(),
+    name: req.body.name,
+    level: req.body.level,
+    proficiencyBonus: req.body.proficiencyBonus,
+    hp: req.body.hp,
+    hpMax: req.body.hpMax,
+    speed: req.body.speed,
+    description: req.body.description,
+    abilityScores: req.body.abilityScores,
+    race: req.body.raceId,
+    class: req.body.characterClassId
+
+  });
+  character.save()
+    .then(result => {
+      res.status(201).json({
+        createdCharacter: {
+          _id: result._id,
+          name: result.name,
+          level: result.level,
+          proficiencyBonus: result.proficiencyBonus,
+          hp: result.hp,
+          hpMax: result.hpMax,
+          speed: result.speed,
+          description: result.description,
+          abilityScores: result.abilityScores,
+          race: result.raceId,
+          class: result.characterClassId,
+          request: {
+            type: 'GET',
+            url: 'http://localhost:3000/characters/' + result._id
+          }
+        }
+      });
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({ error: err });
+    });
+});
+
+router.get('/:charactertId', (req, res, next) => {
+  const id = req.params.charactertId;
+  Character.findById(id)
+    .select('_id name level proficiencyBonus hp hpMax speed description abilityScores race class')
```

Comment:
Is it necessary to write all fields, when we want everything? isn't there a shorthand for selecting everything?

Reply:
![image](https://user-images.githubusercontent.com/12697119/74263623-7ade9380-4cff-11ea-8e60-7f1bc480ac4a.png)
Like this in the mongoDB documentation

Reply:
There is a field __v in the object, that i dont really want to display, is there a way to just specify what i dont want to see?

Reply:
I see that you figured it out by adding `.select('-__v')` instead 👍